### DISCUSSION TITLE
Future Payment - Setting up Payment Method with Flutter

### DISCUSSION THREAD

**Discussion by sk92129** [flutter-stripe/flutter_stripe#2008] [Q&A]
**Describe the bug**
I have my java springboot server create the "create-setup-intent" and returns the clientSecret and customerId.

Here is a snippet of the logs:
I/flutter (19491):  url :: http://MyserverIP:8080/payment/create-setup-intent :: 
I/flutter (19491):  value :: {clientSecret: seti_1QTdC2KKI1hVOxLkYg328iJV_secret_RMLtEVrhFMpLubyXLZgTa2zy6LMT7QY, customerId: cus_RMLtOk04Iyv2Ef} 

When the flutter app get this, i was temporarily able to display the payment bottom sheet for a few times.

But now it stopped.

This is now the error that I see as a toast message.

![image](https://github.com/user-attachments/assets/2539611e-e278-40d0-b693-5f099d9ee0c2)

![stripe-payment-method](https://github.com/user-attachments/assets/81c12dc8-2516-4ad4-9b62-618a5e56db90)

I was using the code from the example "add_payment_method_button.dart" and the related files.  
Since my flutter app is using BLOC, I fetch the clientSecret and customerId from the server and pass those two strings into AddPaymentMethodScreenLoader

Here is the mobile version

![image](https://github.com/user-attachments/assets/84921407-28bb-4246-a36e-c9076e09f131)

**To Reproduce**
1. ) I fetch the clientSecret and customerId from my springboot server.
2. ) I pass those two values to the AddPaymentMethodButton class from the example.
3. 

**Expected behavior**
I expect the payment method entry bottom sheet to appear consistently.  Now it fails about 99% of the time.

**Smartphone / tablet**
 - Device: Android Pixel Simulator 
 - OS: API 35
 - Package version: flutter 
 - Flutter version 3.24.5

**Additional context**

**Comment by sk92129** [comment]
looking at the clientSecret carefully, it looks like two parts concatenated together.
Is this correct?

seti_1QTdC2KKI1hVOxLkYg328iJV_secret_RMLtEVrhFMpLubyXLZgTa2zy6LMT7QY

I am putting the clientSecret and the customerId into a java POJO object to send back via springboot.

        SetupIntent intent = StripeCustomer.createSetupIntent(stripeApiKey);
        SetupIntentResponse response = new SetupIntentResponse();
        response.setClientSecret(intent.getClientSecret());
        response.setCustomerId(stripeCustomerId);
        return response ;

**Comment by remonh87** [comment]
Make sure that you use the clientsecret that you get from the backend for your setupintent. Do not use the stripe publishablekey for it

**Reply by sk92129** [reply]
To confirm that I was doing the correct set up and steps, I went thru your example again.  
The example app for iOS and Android for "payment sheet: setup future payments" does not work either.  
The android app: 

Using LadyBug, and my JDK locally having JDK21, I had to fix Android to work on Android15:

     compileSdk 34
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_17
-        targetCompatibility JavaVersion.VERSION_17
+        sourceCompatibility JavaVersion.VERSION_1_8 
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    
+    kotlinOptions {
+        jvmTarget = JavaVersion.VERSION_1_8 
     }

Otherwise, I got this error: 
https://stackoverflow.com/questions/79012911/unknown-kotlin-jvm-target-21-flutter-build-error-in-beginner-project

**Reply by remonh87** [reply]
there is an issue with ladybug currently that we will look into. I had to revert to the previous android studio. I hope to have time to look into it but it will be challenging around the holidays
